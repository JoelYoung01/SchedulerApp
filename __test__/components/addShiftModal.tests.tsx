import React, { useEffect } from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
