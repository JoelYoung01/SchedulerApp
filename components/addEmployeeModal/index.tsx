export * from "./AddEmployeeModal";
export * from "./AvailiabilityTabs";
export * from "./AvailabilityEditor";
export * from "./EditEmployeeInfo";
export * from "./TabPanel";
