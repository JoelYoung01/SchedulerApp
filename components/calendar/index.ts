export * from "./Calendar";
export * from "./DailyShifts";
export * from "./ShiftCard";
export * from "./ShiftInfo";
export * from "./WeeklyDate";
