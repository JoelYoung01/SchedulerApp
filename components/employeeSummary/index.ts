export * from "./EmployeeSummary";
export * from "./EmployeeCard";
