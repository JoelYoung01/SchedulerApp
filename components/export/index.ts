export * from "./ExportModal";
export * from "./FormatSelect";
