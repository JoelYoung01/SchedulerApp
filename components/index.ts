export * from "./calendar";
export * from "./export";
export * from "./layout";
