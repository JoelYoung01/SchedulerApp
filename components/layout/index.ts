export * from "./DefaultLayout";
export * from "./NavBar";
