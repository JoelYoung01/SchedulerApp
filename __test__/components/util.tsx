import { expect } from "@jest/globals";
import type { MatcherFunction } from "expect";

import "@testing-library/jest-dom";
import { render, screen, fireEvent, Matcher } from "@testing-library/react";

import React from "react";
import { ExportModal, ExportType } from "../../components";

const toBeHidden: MatcherFunction<[]> = function toBeHidden(actual) {
  if (!(actual instanceof HTMLElement)) {
    throw new Error("actual must be an HTMLElement");
  }
  let current: HTMLElement | null = actual;
  while (current !== null) {
    if (current.style.display === "none") {
      return {
        message: () =>
          "Expected element or parent not to be hidden by `style.display = 'none'`",
        pass: true
      };
    }
    current = current.parentElement;
  }
  return {
    message: () =>
      "Expected element or parent to be hidden by `style.display = 'none'`",
    pass: false
  };
};
expect.extend({ toBeHidden });

declare module "expect" {
  interface AsymmetricMatchers {
    toBeHidden: () => void;
  }
  interface Matchers<R> {
    toBeHidden: () => R;
  }
}

export function checkExportHide(
  name: string,
  element: JSX.Element,
  selector: Matcher
): void {
  test(name, async() => {
    const exportRef: React.RefObject<React.ReactInstance> = React.createRef();
    // Need to add export modal to add printing events
    render(
      <>
        {element}
        <ExportModal
          componentToExport={exportRef}
          defaultValue={ExportType.pdf}
        />
        <div className="printed" data-testid="shown" />
      </>
    );
    expect(screen.getByText(selector)).not.toBeHidden();
    fireEvent(window, new Event("beforeprint"));
    expect(screen.getByText(selector)).toBeHidden();
    expect(screen.getByTestId("shown")).not.toBeHidden();
    fireEvent(window, new Event("afterprint"));
    expect(screen.getByText(selector)).not.toBeHidden();
  });
}
