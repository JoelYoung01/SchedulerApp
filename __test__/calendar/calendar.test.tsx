import { Calendar } from "../../components";
import renderer from "react-test-renderer";

test("Test Calendar Renders", () => {
  const calendar = renderer.create(<Calendar allShifts={[]} />);
  expect(calendar).toMatchSnapshot();
});
