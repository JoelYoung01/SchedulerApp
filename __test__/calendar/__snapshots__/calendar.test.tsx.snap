// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calendar Renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-17qsxia-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Sun
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Mon
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Tue
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Wed
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Thu
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Fri
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-dqfxkm-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-16taog-MuiTypography-root"
        >
          10.23
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1osj8n2-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignRight css-1js305m-MuiTypography-root"
        >
          Sat
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1i5d9ma-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1idn90j-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
